iceoryx_posh/source/roudi/memory/iceoryx_roudi_memory_manager.cpp

```diff
@@ -47,7 +47,7 @@ cxx::expected<RouDiMemoryManagerError> IceOryxRouDiMemoryManager::destroyMemory(
 
 PosixShmMemoryProvider* IceOryxRouDiMemoryManager::mgmtMemoryProvider() noexcept
 {
-    return &m_defaultMemory.m_managementShm;
+    return const_cast<PosixShmMemoryProvider*>(const_cast<const IceOryxRouDiMemoryManager*>(this)->mgmtMemoryProvider());
```

Comment:
Avoid duplication code and error link in unit test. Indeed with previous code, even if we are calling const method the non-const was called.

Reply:
I'm all for reducing code duplication, but this makes the code harder to read and and merely reduces code duplication.
What do the others think?

I also do not see how this can solve the issue you described. The signature didn't change. Maybe the underlying problem is masked with this changes.

Reply:
We use the same technique for instance in the optional and here I would argue, even when the method is simplistic I would still use the double const cast. The reason is when  `mgmtMemoryProvider` contains a bug and we fix it we cannot overlook the const overload since we are forwarding from here.

If we have to identical implementations we have to do all changes in two places and this will lead to screw ups...

Reply:
I also don't like the idiom with the const cast very much. It just happens that I like code duplication even less. The difference is just a plank length, though. Unfortunately I haven't found a better solution to prevent code duplication, yet.

Reply:
This is the Scott Meyers strategy.
The inner cast should be a `static_cast` though.